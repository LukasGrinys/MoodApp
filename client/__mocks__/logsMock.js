export const testLogs = [
    {
        id: '001',
        rating: 10,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '002',
        rating: 9,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '003',
        rating: 8,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '004',
        rating: 7,
        date: "TEST-DATE",
        text: "Test text"
    }
];

export const testLog = {
    id: '001',
    rating: 10,
    timing: 'Afternoon', 
    date: "TEST-DATE",
    text: "Test text"
}