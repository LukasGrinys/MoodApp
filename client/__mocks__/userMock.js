export const testUser = {
    id: '001',
    firstName: 'test',
    lastName: 'test',
    email: 'test@test.com'
}