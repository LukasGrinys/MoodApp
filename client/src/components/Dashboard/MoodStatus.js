import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import LoadingNetItem from '../Loading/loadingNetItem';
import { useTheme } from '../../contexts/ThemeContext';
import Button from '../Button/Button';
import parentClasses from './Dashboard.module.scss';
import styles from './MoodStatus.module.scss';
import { useMoodStatus } from '../../hooks/Dashboard/useMoodStatus';
import classNames from 'classnames';

const MoodStatus = () => {
    const darkTheme = useTheme();

    const {
        rating,
        ratingText,
        isFetchingLogs
    } = useMoodStatus();

    return (
        <div className={classNames(
            parentClasses.gridItem,
            styles.statusContainer
        )}>
            {
                !isFetchingLogs ? (
                    <Fragment>
                        <h3 className={parentClasses.gridItemHeader}>Latest mood status</h3>
                        <div className={classNames(
                            styles.statusBox,
                            darkTheme && styles.dark
                        )}>{rating}</div>
                        <div className={styles.statusText}>{ratingText}</div>
                        <Link to="/stats">
                            <Button>Stats</Button> 
                        </Link>
                    </Fragment>
                ) : (
                    <LoadingNetItem/>
                )
            }
        </div>
    )
};

export default MoodStatus;