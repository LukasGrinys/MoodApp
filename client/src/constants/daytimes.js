export const daytimes = {
    morning: 'Morning',
    afternoon: 'Afternoon',
    evening: 'Evening',
    night: 'Night'
}