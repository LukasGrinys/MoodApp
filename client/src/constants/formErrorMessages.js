export const formErrorMessages = {
    required : 'Required field',
    email: 'Enter a proper e-mail address'
}