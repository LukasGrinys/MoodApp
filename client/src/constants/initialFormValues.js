export const initialFormValues = {
    email: '',
    password: '',
    firstName: '',
    lastName: '',
    confirmPassword: '',
    currentPassword: ''
}