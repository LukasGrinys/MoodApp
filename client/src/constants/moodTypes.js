const moodTypes = [
    {
        above: 7.5,
        below: 10,
        text: 'Great'
    },
    {
        above: 5,
        below: 7.5,
        text: 'Good'
    },
    {
        above: 3,
        below: 5,
        text: 'Average'
    },
    {
        above: 0,
        below: 3,
        text: 'Bad'
    }
];

export default moodTypes;