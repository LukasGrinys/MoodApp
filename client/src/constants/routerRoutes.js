export const routerRoutes = {
    home: '/',
    login: '/login',
    logout: '/logout',
    dashboard: '/dashboard',
    addNewLog: '/addlog',
    register: '/register',
    settings: '/settings'
}