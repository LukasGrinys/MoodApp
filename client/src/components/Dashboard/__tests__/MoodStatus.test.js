import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import MoodStatus from '../MoodStatus';
import LoadingNetItem from '../../Loading/loadingNetItem';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { routerRoutes } from '../../../constants/routerRoutes';
import { testLogs } from '../../../../__mocks__/logsMock';

const mockStore = configureStore();

describe('MoodStatus: loading', () => {
    const initialState = {
        user: {},
        logs: {
            lastLogs : {
                isFetching: true,
                logs: null
            }
        }
    }

    const store = mockStore(initialState);

    const wrapper = mount(
        <Provider store={store}>
            <MoodStatus/>
        </Provider>
    );

    test('Renders properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Renders loading widget', () => {
        expect(wrapper.find(LoadingNetItem).length).toBe(1);
    })
});

describe('MoodStatus without any logs', () => {
    const store = mockStore({
        user: {},
        logs: {
            lastLogs : {
                isFetching: false,
                logs: []
            }
        }
    });

    const wrapper = mount(
        <Router>
            <Provider store={store}>
                <MoodStatus/>
            </Provider>
        </Router>
    );

    test('Renders properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Has a status box', () => {
        expect(wrapper.find('.statusBox').length).toBe(1);
    });

    test('Status box shows N/A instead of rating', () => {
        expect(wrapper.find('.statusBox').text()).toBe('N/A');
    });

    test('Has a link to stats section', () => {
        expect(wrapper.find('Link').length).toBe(1);
        expect(wrapper.find('Link').prop('to')).toBe(routerRoutes.stats);
    });
});

describe('MoodStatus with logs', () => {
    const store = mockStore({
        user: {},
        logs: {
            lastLogs : {
                isFetching: false,
                logs: testLogs
            }
        }
    });

    const wrapper = mount(
        <Router>
            <Provider store={store}>
                <MoodStatus/>
            </Provider>
        </Router>
    );

    test('Renders properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Has a status box', () => {
        expect(wrapper.find('.statusBox').length).toBe(1);
    });

    test('Status box shows rating', () => {
        /* Here at component level we assume that average of
        test logs is 9 for now */;
        expect(wrapper.find('.statusBox').text()).toBe('9.0');
    });

    test('Has a status text', () => {
        /* This test already indicates that we should export the
        text generating function as a helper */
        expect(wrapper.find('.statusText').length).toBe(1);
        expect(wrapper.find('.statusText').text()).toBe('Great');
    })

    test('Has a link to stats section', () => {
        expect(wrapper.find('Link').length).toBe(1);
        expect(wrapper.find('Link').prop('to')).toBe(routerRoutes.stats);
    });
});

