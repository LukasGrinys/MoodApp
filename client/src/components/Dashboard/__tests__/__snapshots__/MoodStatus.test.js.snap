// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoodStatus with logs Renders properly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <MoodStatus>
        <div
          className="gridItem statusContainer"
        >
          <h3
            className="gridItemHeader"
          >
            Latest mood status
          </h3>
          <div
            className="gridItemContent"
          >
            <div
              className="statusBox"
            >
              9.0
            </div>
            <div
              className="statusText"
            >
              Great
            </div>
            <Link
              to="/stats"
            >
              <LinkAnchor
                href="/stats"
                navigate={[Function]}
              >
                <a
                  href="/stats"
                  onClick={[Function]}
                >
                  <Button>
                    <button
                      className="button btnPrimary"
                      onClick={[Function]}
                      type="button"
                    >
                      Stats
                    </button>
                  </Button>
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </div>
      </MoodStatus>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`MoodStatus without any logs Renders properly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <MoodStatus>
        <div
          className="gridItem statusContainer"
        >
          <h3
            className="gridItemHeader"
          >
            Latest mood status
          </h3>
          <div
            className="gridItemContent"
          >
            <div
              className="statusBox"
            >
              N/A
            </div>
            <div
              className="statusText"
            />
            <Link
              to="/stats"
            >
              <LinkAnchor
                href="/stats"
                navigate={[Function]}
              >
                <a
                  href="/stats"
                  onClick={[Function]}
                >
                  <Button>
                    <button
                      className="button btnPrimary"
                      onClick={[Function]}
                      type="button"
                    >
                      Stats
                    </button>
                  </Button>
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </div>
      </MoodStatus>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`MoodStatus: loading Renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MoodStatus>
    <div
      className="gridItem statusContainer"
    >
      <LoadingNetItem>
        <div
          className="loadingScreen"
        >
          <div
            className="barContainer"
          >
            <div
              className="bar"
            />
            <div
              className="bar delay1"
            />
            <div
              className="bar delay2"
            />
          </div>
        </div>
      </LoadingNetItem>
    </div>
  </MoodStatus>
</Provider>
`;
